joblib/test/test_memory.py

```diff
@@ -722,3 +724,22 @@ def test_memory_clear(tmpdir):
     memory.clear()
 
     assert os.listdir(memory.cachedir) == []
+
+
+def fast_func_with_complex_output():
+    complex_obj = ['a' * 1000] * 1000
+    return complex_obj
+
+
+@with_multiprocessing
+def test_cached_function_race_condition_when_persisting_output(tmpdir, capfd):
+    memory = Memory(cachedir=tmpdir.strpath)
+    func_cached = memory.cache(fast_func_with_complex_output)
+
+    Parallel(n_jobs=2)(delayed(func_cached)() for i in range(10))
+
+    stdout, stderr = capfd.readouterr()
+
+    exception_msg = 'Exception while loading results'
+    assert exception_msg not in stdout
```

Comment:
Checking both stdout and stderr is done with #434 in view (in which case the warning will be on stdout) and to prevent the false security of a test not actually testing anything.